### PR TITLE
Feature/city cure player tests

### PR DESCRIPTION
**Pull request by IsaacSShade**
Player cannot be tested that well until we get the gameboard object. I implemented some bug fixes for issues I found while doing these tests.

### CODE REVIEW

**Review by tonixsmm — APPROVED** [review]

File: src/test/java/edu/gonzaga/CityTest.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    void cityResearchStation1() {
+        Boolean desiredValue = false;
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.removeResearchStation();
+
+        Assertions.assertEquals(desiredValue, testCity1.getResearchStation());
+    }
+
+    /**
```
Comment (izzytilles) [inline src/test/java/edu/gonzaga/CityTest.java:160]: Would be helpful to add comment to test that specifies that you also need to read console output (!! Attempt to add a research station to a city that already has one !!) to make sure it won't double-add

File: src/test/java/edu/gonzaga/CityTest.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+     * Tests that a city can't double add a research station
```
Comment (izzytilles) [inline src/test/java/edu/gonzaga/CityTest.java:161]: Same as earlier comment, test requires reading console to make sure function works

File: src/main/java/edu/gonzaga/City.java [hunk 3]
```diff
@@ -212,6 +217,24 @@ public Color getColor() {
...
+     * 
+     * @author Aiden T
+     */
+    public void outbreakCleanup() {
+        if (this.getInfectionCubes().size() > 3) {
+            removeInfectionCube(cityColor);
+        }
+
+        for (City city : this.connections) {
+            if (city.getInfectionCubes().size() > 3) {
```
Comment (izzytilles) [inline src/main/java/edu/gonzaga/City.java:231]: ^^ magic number

File: src/main/java/edu/gonzaga/City.java [hunk 4]
```diff
@@ -130,9 +137,7 @@ public Integer addInfectionCube(Color color) {
         if (this.infectionCubes.size() == 4) {
```
Comment (izzytilles) [inline src/main/java/edu/gonzaga/City.java:137]: magic num

File: src/main/java/edu/gonzaga/City.java [hunk 5]
```diff
@@ -130,9 +137,7 @@ public Integer addInfectionCube(Color color) {
...
 
             outbreakCount = this.outbreak();
-            this.infectionCubes.remove(color);
             
-
         } else if (this.infectionCubes.size() > 4){
```
Comment (izzytilles) [inline src/main/java/edu/gonzaga/City.java:141]: magic num

**Review by izzytilles — APPROVED** [review]
Everything looks great and the tests are thorough, just 2 comment elaboration requests on tests and a few magic numbers found on older code (inspired by class today)

**Pull request merged**